vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher_test.go

```diff
@@ -117,6 +117,37 @@ func TestControllerFetcher(t *testing.T) {
 				Name: "test-deployment", Kind: "Deployment", Namespace: "test-namesapce"}}, // Deployment has no parent
 			expectedError: nil,
 		},
+		{
+			key: &ControllerKeyWithAPIVersion{ControllerKey: ControllerKey{
+				Name: "test-ss", Kind: "StatefulSet", Namespace: "test-namesapce"}},
+			objects: []runtime.Object{&appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "Deployment",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-deployment",
+					Namespace: "test-namesapce",
+				},
+			}, &appsv1.StatefulSet{
```

Comment:
I guess this is OK for test purposes, but it is kind of weird to see stateful set controlled by a deployment. Maybe it would make more sense to make it standalone, e.g. it has no parent?

Reply:
I wrote just as ReplicaSet case did. Okay, I will try to figure it out :)